ocular/src/chain/client/automated_tx_handler.rs

```diff
@@ -0,0 +1,462 @@
+#![warn(unused_qualifications)]
+#![allow(unused_imports)]
+// Clippy broken; doesn't recognize certain imports are used and sees them as unused
+
+use crate::{
+    chain::{
+        client::tx::{Account, TxMetadata},
+        config::ChainClientConfig,
+    },
+    error::AutomatedTxHandlerError,
+    keyring::Keyring,
+};
+use bip32::Mnemonic;
+use cosmrs::{bank::MsgSend, rpc, tx::Msg, AccountId, Coin};
+
+use serde::{Deserialize, Serialize};
+use std::time::SystemTime;
+use std::{fs, path::Path, str::FromStr};
+use tendermint_rpc::endpoint::broadcast::tx_commit::Response;
+use uuid::Uuid;
+
+use super::ChainClient;
+
+// Toml structs
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct DelegatedToml<'a> {
+    pub sender: DelegatedSender<'a>,
+
+    #[serde(borrow)]
+    pub transaction: Vec<DelegateTransaction<'a>>,
+}
+
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct DelegatedSender<'a> {
+    pub source_private_key_path: &'a str,
+    pub delegate_expiration_unix_seconds: i64,
+    // Feegrant data
+    pub fee_grant_expiration_unix_seconds: i64,
+    pub fee_grant_amount: u64,
+    pub denom: &'a str,
+    // MsgGrant data
+    // TODO: Remove account and sequence numbers. Get automatically from account type via https://github.com/PeggyJV/ocular/issues/25
+    pub grant_account_number: u64,
+    pub grant_sequence_number: u64,
+    pub grant_gas_fee: u64,
+    pub grant_gas_limit: u64,
+    pub grant_timeout_height: u32,
+    pub grant_memo: &'a str,
+    // MsgExec data
+    pub exec_gas_fee: u64,
+    pub exec_gas_limit: u64,
+    pub exec_timeout_height: u32,
+    pub exec_memo: &'a str,
+}
```

Comment:
I'm open to alternative sender object design here. This is a bit verbose and depending on our needs/wants we can potentially simplify this a bit.